solidity/revert-strings-to-custom-error-migration.md

```diff
@@ -0,0 +1,25 @@
+# Purpose
+
+The purpose of this document is to standardize all error messages/handling for the Optimism Smart Contracts, which currently use a mix of error strings, and mix both `require` and `revert`.
+
+# Summary
+
+Going forward, all error messages should use the prefix of the `contract` name and the suffix of `*Error`. Additionally, all errors should be returned via `revert` rather than `require`. This is both for visual clarity and due to the efficiency concerns around `require` in the case of errors with arguments.
```

Comment:
Can you provide an example?

Reply:
Added in a clear example [here](https://github.com/ethereum-optimism/design-docs/pull/143/commits/8f67890726d1fa5cd4848107a51d9d5b0b7a88b2)